tls/s2n_ecc_preferences.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+
+#include "tls/s2n_ecc_preferences.h"
+#include "tls/s2n_connection.h"
+#include "crypto/s2n_ecc_evp.h"
+#include "utils/s2n_safety.h"
+
+const struct s2n_ecc_named_curve *const s2n_ecc_pref_list_20140601[] = {
+    &s2n_ecc_curve_secp256r1,
+    &s2n_ecc_curve_secp384r1,
+};
+
+const struct s2n_ecc_named_curve *const s2n_ecc_pref_list_20200310[] = {
+    &s2n_ecc_curve_secp256r1,
+    &s2n_ecc_curve_secp384r1,
+#if MODERN_EC_SUPPORTED
+    &s2n_ecc_curve_x25519,
+#endif
+};
+
```

Comment:
if we are only sending the first keyshare, i would like a ecc curve preference that has x25519 on top.

Reply:
Was this a miss, or is x25519 priority changed?

Reply:
I have updated the list to have` x25519` as the first priority.